packages/react-components/src/components/tooltip-xy-component.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+type MaybePromise<T> = T | Promise<T>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface TooltipXYComponentState<T = any> {
+    tooltipData: T
+}
+
+export class TooltipXYComponent extends React.Component<unknown, TooltipXYComponentState> {
+
+    constructor(props: unknown) {
+        super(props);
+        this.state = {
+            tooltipData: undefined
+        };
+    }
+
+    render(): React.ReactNode {
+        return (
+            <div style={{
+                padding: 10,
+                position: 'fixed',
+                textAlign: 'left',
+                color: 'white',
+                border: '1px solid transparent',
+                backgroundColor: 'rgba(51, 122, 183, 0.9)',
+                borderRadius: 3,
+                top: this.state.tooltipData?.top,
+                left: this.state.tooltipData?.left,
+                opacity: this.state.tooltipData?.opacity ? this.state.tooltipData.opacity : 0,
+                zIndex: 999
+                }}
+            >
+                <p style={{margin: '0 0 5px 0'}}>{this.state.tooltipData?.title}</p>
+                <ul style={{padding: '0'}}>
+                    {this.state.tooltipData?.dataPoints?.map((point: { color: string, background: string, label: string, value: number }, i: number) =>
+                        <li key={i} style={{listStyle: 'none', display: 'flex', marginBottom: 5}}>
+                            <div style={{
+                                height: '10px',
+                                width: '10px',
+                                margin: 'auto 0',
+                                border: 'solid thin',
+                                borderColor: point.color,
+                                backgroundColor: point.background
+                                }}
+                            ></div>
+                            <span style={{marginLeft: '5px'}}>{point.label} {point.value}</span>
+                        </li>
+                    )}
+                </ul>
+                {this.state.tooltipData?.zeros > 0 &&
+                    <p style={{marginBottom: 0}}>{this.state.tooltipData?.zeros} other{this.state.tooltipData?.zeros > 1 ? 's' : ''}: 0</p>
+                }
+            </div>
+        );
+    }
+
+    setElement<T>(tooltipData: T): void {
+        this.setState({ tooltipData });
```

Comment:
what will the display look like if there a large number of non-zero lines? Will the tooltip window just grow?
We could limit the size and create a scrollbar (or more button). But this would require that the user is able to click into the tooltip and use the scrollbar.

Reply:
Done.